java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -968,6 +971,42 @@ public static Table readJSON(Schema schema, JSONOptions opts, byte[] buffer) {
     return readJSON(schema, opts, buffer, 0, buffer.length);
   }
 
+  private static Table gatherJSONColumns(Schema schema, TableWithMeta twm) {
+    String[] neededColumns = schema.getColumnNames();
+    if (neededColumns == null || neededColumns.length == 0) {
+      return twm.releaseTable();
+    } else {
+      String[] foundNames = twm.getColumnNames();
+      HashMap<String, Integer> indices = new HashMap<>();
+      for (int i = 0; i < foundNames.length; i++) {
+        indices.put(foundNames[i], i);
+      }
```

Comment:
I just don't see it as better here. 3 lines instead of 4. I personally find the for loop simpler to read. Because the number of columns is likely to be small I don't really want to get into performance, but I think the for loop is going to be faster too because there is a lot less overhead with it.